File: src/main/java/com/jcfp/tallererp/entity/OrdenTrabajo.java
package com.jcfp.tallererp.entity;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import java.util.List;
import lombok.Getter;
import lombok.Setter;
import java.time.LocalDate;

@Entity(name = "ordenes")
@Getter
@Setter
public class OrdenTrabajo{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String codigoOrden;
    @ManyToOne
    private Empleado empleadoAsignado;
    @ManyToOne
    private Vehiculo vehiculo;
    @OneToMany(mappedBy = "ordenTrabajo", cascade = CascadeType.PERSIST)
    @JsonManagedReference
    private List<ArticuloUsado> articulosUsados;
    @ManyToOne
    private EstadoOrden estadoOrden;
    private LocalDate fechaOrden;
    private LocalDate fechaInicio;
    private LocalDate fechaFinalizacion;
    private LocalDate fechaPago;
    private boolean pagada;
    private String observaciones;
    private double total;
}

File: frontend/src/types/entities/orden_trabajo.ts
import type { Empleado } from "./empleado";
import type { Vehiculo } from "./vehiculo";

export interface EstadoOrden {
  id: string;
  nombre: string;
}

export interface OrdenTrabajo {
  id: string;
  codigoOrden: string;
  empleadoAsignado: Empleado | { id: string; nombre: string };
  vehiculo: Vehiculo | { id: string; matricula: string };
  articulosUsados: ArticuloUsado[];
  estadoOrden: EstadoOrden;
  fechaOrden: string;
  fechaInicio?: string;
  fechaFinalizacion?: string;
  fechaPago?: string;
  pagada: boolean;
  observaciones?: string;
  total: number;
}

export interface OrdenTrabajoPayload {
  codigoOrden: string;
  empleadoAsignado: {
    id: string;
  };
  vehiculo: {
    id: string;
  };
  articulosUsados: ArticuloUsado[];
  estadoOrden: {
    id: string;
  };
  fechaOrden: string;
  fechaInicio: string;
  fechaFinalizacion: string;
  fechaPago: string;
  pagada: boolean;
  observaciones: string;
  total: number;
}

export interface OrdenTrabajoEditData extends OrdenTrabajoPayload {
  id: string;
}

export interface ArticuloUsado {
  id?: string;
  articulo: {
    id: string;
    descripcion: string;
    precio: number;
    stock?: number;
  };
  cantidad: number;
  ordenTrabajo?: {
    id: string;
  };
}
